services/indexes/pvm/writer.go

```diff
@@ -339,11 +347,11 @@ func initializeTx(version uint16, c codec.Manager, tx platformvm.Tx) error {
 }
 
 func (w *Writer) indexBlock(ctx services.ConsumerCtx, proposerblockBytes []byte) error {
-	var pblock platformvm.Block
+	var pblock blocks.Block
 	var ver uint16
 	var err error
 
-	proposerBlock, err := block.Parse(proposerblockBytes)
+	proposerBlock, _, err := block.Parse(proposerblockBytes)
 	blockBytes := proposerblockBytes
 	if err == nil {
```

Comment:
Explicitly using `codec.Unmarshal` is discouraged. You should use `blocks.Parse(codec, bytes)`: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/blocks/parse.go#L10

Reply:
Sure, what is the difference between a Blocks.block object and a Block.block object.  All of the following code seems to be predicated on a block type